### DISCUSSION TITLE
Is it possible to use useMenuState hook on Popover?

### DISCUSSION THREAD

**Discussion by irangarcia** [ariakit/ariakit#1396] [Questions] [answered]
In my use case, we have a Custom Popover that uses the ariakit one, to allow the menu navigation, other a11y features as menu/menuitem role and to avoid duplicating the styles creating another component, we tried to pass the `useMenuState` to it but we didn't achieve it. There is a way to do that?

simple snippet: 
```tsx
const menu = useMenuState()

....

<CustomPopover state={menu} ...>
    ...
</Popover>
```

**Comment by diegohaz** [comment, marked as answer]
> to allow the menu navigation, other a11y features as menu/menuitem role

Those features aren't provided by `useMenuState`. State hooks don't add any semantics to the elements. Components do.

If you want menu navigation and `menu`/`menuitem` roles, you should use `Menu` and `MenuItem` components.

**Comment by marcospassos** [comment]
@diegohaz, I think what @irangarcia meant was how to use a custom popover with the menu since it uses the popover underneath. 

All popovers share the same style in our design system, so we built a custom`<Popover />` component that specifies not only style but also animation. Using the same component, we can ensure consistency across the DS and eliminate duplicate code. However, we found no way to wire the menu and popover for composability.

Do you have any suggestions?

**Reply by diegohaz** [reply]
My suggestion is that you abstract your styles into a separate component, so you can use it with `Menu`, `Popover` etc.

That's what we're doing on the website:

https://github.com/ariakit/ariakit/blob/da08e9809510b873be5c76122dd240966d9ee504/packages/website/components/popup/index.tsx#L1-L35

https://github.com/ariakit/ariakit/blob/fc0a669c8c5a07403386b5e027765dc343e85598/packages/website/components/playground/index.tsx#L198-L205
